Storage/Logins.swift

```diff
@@ -522,7 +527,7 @@ open class Login: CustomStringConvertible, LoginData, LoginUsageData, Equatable
             }
         }
 
-        let out = Login(guid: guid, hostname: hostname, username: username!, password: password)
+        let out = Login(guid: guid, hostname: hostname, username: username ?? "", password: password)
```

Comment:
I don't know why `username` is an optional and `password` is not.

Reply:
Agreed 100x. This is how the original code is, I'll take another look and see if I can cleanup this code a bit.

Reply:
The code is trying to indicate that if a username is set, then a password is expected. I don't find that too clear myself.
I think either both should be optional, or both should be non-optional and empty string means not set. Any preference? Either one results in a cascade of code changes.
Having just typed those suggestions, I am thinking now that leaving it is the best option until someone is doing more refactoring in there.